File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/image/ImagePersistenceAdapter.java
package kyonggiyo.persistence.image;

import kyonggiyo.application.image.port.outbound.DeleteImagePort;
import kyonggiyo.application.image.port.outbound.LoadImagePort;
import kyonggiyo.application.image.port.outbound.SaveImagePort;
import kyonggiyo.application.image.domain.entity.Image;
import kyonggiyo.application.image.domain.vo.ImageType;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@RequiredArgsConstructor
public class ImagePersistenceAdapter implements SaveImagePort, LoadImagePort, DeleteImagePort {

    private final ImageRepository imageRepository;
    private final ImageBatchRepository imageBatchRepository;

    @Override
    public List<Image> saveAll(List<Image> images) {
        return imageRepository.saveAll(images);
    }

    @Override
    public void saveAllInBatch(List<Image> images) {
        imageBatchRepository.saveAllInBatch(images);
    }

    @Override
    public List<Image> findAllByImageTypeAndReferenceId(ImageType imageType, Long referenceId) {
        return imageRepository.findByImageTypeAndReferenceId(imageType, referenceId);
    }

    @Override
    public List<Image> findAllByImageTypeAndReferenceIdIn(ImageType imageType, List<Long> ids) {
        return imageRepository.findByImageTypeAndReferenceIdIn(imageType, ids);
    }

    @Override
    public void deleteById(Long id) {
        imageRepository.deleteById(id);
    }

    @Override
    public void deleteAllByIdInBatch(List<Long> ids) {
        imageRepository.deleteAllByIdInBatch(ids);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/image/adapter/ImagePersistenceAdapter.kt
package kyonggiyo.image.adapter

import kyonggiyo.application.image.domain.entity.Image
import kyonggiyo.application.image.domain.vo.ImageType
import kyonggiyo.application.image.port.outbound.DeleteImagePort
import kyonggiyo.application.image.port.outbound.LoadImagePort
import kyonggiyo.application.image.port.outbound.SaveImagePort
import kyonggiyo.image.repository.ImageBatchRepository
import kyonggiyo.image.repository.ImageRepository
import org.springframework.stereotype.Component

@Component
class ImagePersistenceAdapter(
        private val imageRepository: ImageRepository, 
        private val imageBatchRepository: ImageBatchRepository
) : SaveImagePort, LoadImagePort, DeleteImagePort {
   
    override fun saveAll(images: List<Image>): List<Image> {
        return imageRepository.saveAll(images)
    }

    override fun saveAllInBatch(images: List<Image>) {
        imageBatchRepository.saveAllInBatch(images)
    }

    override fun findAllByImageTypeAndReferenceId(imageType: ImageType, referenceId: Long): List<Image> {
        return imageRepository.findByImageTypeAndReferenceId(imageType, referenceId)
    }

    override fun findAllByImageTypeAndReferenceIdIn(imageType: ImageType, ids: List<Long>): List<Image> {
        return imageRepository.findByImageTypeAndReferenceIdIn(imageType, ids)
    }

    override fun deleteById(id: Long) {
        imageRepository.deleteById(id)
    }

    override fun deleteAllByIdInBatch(ids: List<Long>) {
        imageRepository.deleteAllByIdInBatch(ids)
    }
}
